objstore/objstore.go

```diff
@@ -0,0 +1,137 @@
+package objstore
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"fx.prodigy9.co/config"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+type Client struct {
+	bucket string
+	cfg    *config.Source
+	client *minio.Client
+}
+
+var DefaultClient = &Client{}
+
+var (
+	StorageAccessKeyConfig = config.Str("STORAGE_ACCESS_KEY")
+	StorageSecretKeyConfig = config.Str("STORAGE_SECRET_KEY")
+	StorageEndpointConfig  = config.Str("STORAGE_ENDPOINT")
+	StorageBucketConfig    = config.Str("STORAGE_BUCKET")
+)
+
+func NewClient(cfg *config.Source) *Client {
+	return newS3Client(cfg)
```

Comment:
Functionally equivalent, so no need to have separate `NewClient` vs `newS3Client` here?